File: MyDailyBeat/API/VerveAPI/HealthInfo.m
//
//  HealthInfo.m
//  VerveAPI
//
//  Created by Virinchi Balabhadrapatruni on 5/28/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//

#import "HealthInfo.h"

@implementation HealthInfo

@synthesize uniqueId = _uniqueId;
@synthesize URL = _URL;
@synthesize logoURL = _logoURL;

- (id)initWithUniqueId:(int)uniqueId URL:(NSString *)URL logoURL:(NSString *)logoURL {
    if ((self = [super init])) {
        self.uniqueId = uniqueId;
        self.URL = URL;
        self.logoURL = logoURL;
    }
    return self;
}

- (void)encodeWithCoder:(NSCoder *)coder {
    [coder encodeObject:[NSNumber numberWithInt:self.uniqueId] forKey:@"myHealthPortalUniqueID"];
    [coder encodeObject:self.URL forKey:@"myHealthPortalURL"];
    [coder encodeObject:self.logoURL forKey:@"myHealthPortallogoURL"];
}

- (id)initWithCoder:(NSCoder *)coder {
    self = [super init];
    if (self) {
        self.uniqueId = [(NSNumber *) [coder decodeObjectForKey:@"myHealthPortalUniqueID"] intValue];
        self.URL = (NSString *) [coder decodeObjectForKey:@"myHealthPortalURL"];
        self.logoURL = (NSString *) [coder decodeObjectForKey:@"myHealthPortallogoURL"];
    }
    return self;
}

@end

File: MDB_Swift/MyDailyBeat/API/src/HealthInfo.swift
//  Converted with Swiftify v1.0.6242 - https://objectivec2swift.com/
//
//  HealthInfo.swift
//  VerveAPI
//
//  Created by Virinchi Balabhadrapatruni on 5/28/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//
import Foundation
public class HealthInfo: NSObject, NSCoding {

    public var uniqueId: Int = 0
    public var url: String = ""
    public var logoURL: String = ""

    public init(uniqueId: Int, url URL: String, logoURL: String) {
        self.uniqueId = uniqueId
        self.url = URL
        self.logoURL = logoURL
    }


    public func encode(with coder: NSCoder) {
        coder.encode(self.uniqueId, forKey: "myHealthPortalUniqueID")
        coder.encode(self.url, forKey: "myHealthPortalURL")
        coder.encode(self.logoURL, forKey: "myHealthPortallogoURL")
    }

    required public init?(coder: NSCoder) {
        super.init()
        self.uniqueId = coder.decodeInteger(forKey: "myHealthPortalUniqueID")
        self.url = (coder.decodeObject(forKey: "myHealthPortalURL") as? String)!
        self.logoURL = (coder.decodeObject(forKey: "myHealthPortallogoURL") as? String)!
    
    }
}
